store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,168 @@
+						{Key: model.DevKeyStatus, Value: 1},
+						{Key: model.DevKeyId, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexDevices_Status).
+						SetUnique(false),
+				}, {
+					Keys: bson.D{
+						{Key: model.DevKeyIdDataSha256, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexDevices_IdentityDataSha256).
+						SetUnique(true),
+				},
+			},
+		},
+		DbAuthSetColl: {
+			Indexes: []mongo.IndexModel{
+				{
+					Keys: bson.D{
+						{Key: model.TenantIDField, Value: 1},
+						{Key: model.AuthSetKeyIdDataSha256, Value: 1},
+						{Key: model.AuthSetKeyPubKey, Value: 1},
+					},
+					Options: mopts.Index().
+						SetName(indexAuthSet_IdentityDataSha256_PubKey).
+						SetUnique(true),
+				},
+			},
+		},
+	}
+
+	// for each collection in main deviceauth database...
+	if currentDbName == DbName {
+		for collection, indexModel := range collectionsIndexes {
+			coll := m.ds.client.Database(currentDbName).Collection(collection)
+			// drop all the existing indexes, ignoring the errors
+			_, _ = coll.Indexes().DropAll(m.ctx)
+
+			// create the new indexes
+			if len(indexModel.Indexes) != 0 {
+				_, err := coll.Indexes().CreateMany(m.ctx, indexModel.Indexes)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	// for each collection...
+	for collection := range collectionsIndexes {
+		coll := m.ds.client.Database(currentDbName).Collection(collection)
+		collOut := m.ds.client.Database(DbName).Collection(collection)
+		writes := make([]mongo.WriteModel, 0, findBatchSize)
+
+		if currentDbName == DbName {
+			// if any documents already exist in "deviceauth" db,
+			// add empty "tenant_id": "" key-value pair
+			tenantIdFilter := bson.D{
+				{Key: model.TenantIDField, Value: bson.D{{Key: "$exists", Value: false}}}}
+			update := bson.M{"$set": bson.M{model.TenantIDField: ""}}
+			result, err := collOut.UpdateMany(m.ctx, tenantIdFilter, update)
+			logger.Infof("Modified documents in main deviceauth database, collection %s count: %d",
+				collection, result.ModifiedCount)
+			if err != nil {
+				return err
+			}
+		} else {
+			// get all the documents in the collection
+			findOptions := mopts.Find().
+				SetBatchSize(findBatchSize).
+				SetSort(bson.D{{Key: "_id", Value: 1}})
+			cur, err := coll.Find(m.ctx, bson.D{}, findOptions)
+			if err != nil {
+				return err
+			}
+			defer cur.Close(m.ctx)
+
+			// migrate the documents
+			for cur.Next(m.ctx) {
+				item := bson.D{}
+				err := cur.Decode(&item)
+				if err != nil {
+					return err
+				}
+
+				item = mstore.WithTenantID(m.ctx, item)
+				writes = append(writes, mongo.NewInsertOneModel().SetDocument(item))
+
+				if len(writes) == findBatchSize {
+					_, err := collOut.BulkWrite(m.ctx, writes)
+					if err != nil {
+						return err
+					}
+					writes = writes[:0]
+				}
+			}
+			if len(writes) > 0 {
+				_, err := collOut.BulkWrite(m.ctx, writes)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
```

Comment:
We also need to handle the tokens collection (but separately). We need to make sure that the TTL index exists for the `exp` token claim. See `migration_1_8_0.go`.

Reply:
Done!